pkg/segment/segexecution.go

```diff
@@ -208,6 +208,8 @@ func HelperQueryArithmeticAndLogical(queryOp *structs.QueryArithmetic, resMap ma
 				resultRHS = resMap[queryOp.LHS]
 			}
 
+			// If the query has vector matching operation, then we need to perform operations on that matching label set.
+
```

Comment:
Just to clarify: is this comment correct even if `timeseriesApi == true`? Or is it not possible for both `hasVectorMatchingOp` and `timeseriesApi` to be `true` at the same time?

Reply:
Yes, so for example, when I say `timeseriesApi`, it is the metric query request that comes through our UI. 
Right now we are not allowing to modify the query and users can only make single queries (a, b, c..) and then use formula such as `(a + b + c)` to do some operations. 

but if we let the users to edit these queries and they an write proper promql queries where a single query (a) can include multiple queries.. and in that query, the user can have `on, ignoring` matching label set. 

So in that case if `hasVectorMatchingOp` is true, that should always take priority.